### PR TITLE
 BE-21-37_interview_entity_and_service_tests_for_service

### PR DESCRIPTION
**Pull request by nazfurdychka**

### CODE REVIEW

**Review by AlKaliada — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/project/interview/entity/Interview.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank(message = "Description of interview should not be blank")
+    @Column(name = "additional_info")
+    private String name;
+
+    @OneToOne(cascade = CascadeType.ALL)
```
[the remark this answers is not in the document]
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/interview/entity/Interview.java]: fixed

File: src/main/java/com/exadel/project/internship/dto/InternshipDetailsDTO.java [hunk 2]
```diff
@@ -19,6 +19,7 @@
     private String image;
     private Format format;
     private String internshipType;
+    private String additionalInfoInternship;
```
[the remark this answers is not in the document]
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/internship/dto/InternshipDetailsDTO.java:21]: fixed

File: src/main/java/com/exadel/project/interview/entity/InterviewTime.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "start_date")
+    private LocalDateTime startDate;
+
+    @Column(name = "end_date")
+    private LocalDateTime endDate;
+
+    @OneToOne(mappedBy = "interviewTime")
```
[the remark this answers is not in the document]
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/interview/entity/InterviewTime.java]: fixed

File: src/main/java/com/exadel/project/interview/entity/Interview.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank(message = "Description of interview should not be blank")
+    @Column(name = "additional_info")
+    private String name;
+
+    @ManyToOne
+    private InterviewTime interviewTime;
```
Comment (AlKaliada) [inline src/main/java/com/exadel/project/interview/entity/Interview.java:25]: you lost join column

**Review by AlKaliada — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/project/interview/entity/Interview.java (hunk 4 above)
Comment (nazfurdychka) [reply src/main/java/com/exadel/project/interview/entity/Interview.java:26]: thanks, fixed

**Review by AlKaliada — APPROVED** [review]

**Review by shpaser2 — APPROVED** [review]

**Pull request merged**